adb_shell/tcp_handle.py

```diff
@@ -55,26 +56,35 @@ class TcpHandle(object):
         Default timeout in seconds for TCP packets, or ``None``
     host : str
         The address of the device
-    port : str
+    port : int
         The device port to which we are connecting (default is 5555)
-    serial_number : str
-        ``<host>:<port>``
 
     """
-    def __init__(self, serial, default_timeout_s=None):
-        if ':' in serial:
-            self.host, port = serial.split(':')
-            self.port = int(port)
-        else:
-            self.host = serial
-            self.port = 5555
+    DEFAULT_PORT = 5555
 
-        self.serial = '{}:{}'.format(self.host, self.port)
+    def __init__(self, host, port=DEFAULT_PORT, default_timeout_s=None):
+        self.host = host
+        self.port = port
 
         self._default_timeout_s = default_timeout_s
 
         self._connection = None
 
+    @classmethod
```

Comment:
I think we can get rid of this method and use

```python
def __init(self, host, port=DEFAULT_PORT, default_timeout_s=None):
```

instead of

```python
def __init__(self, serial, default_timeout_s=None):
```

Then instead of checking for a ":" in `serial` and splitting it, we can simply require the user to provide the host and port separately.

Reply:
Yes, the `from_socket_address` classmethod is completely optional, for convenience (and some compatibility with the original API).